File: src/test/java/com/flexpoker/game/command/aggregate/tablebalancer/TableBalancerResumedTableTest.java
package com.flexpoker.game.command.aggregate.tablebalancer;

import static com.flexpoker.game.command.aggregate.tablebalancer.TableBalancerTestUtils.createDefaultChipMapForSubjectTable;
import static com.flexpoker.game.command.aggregate.tablebalancer.TableBalancerTestUtils.createTableToPlayersMap;
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Collections;
import java.util.UUID;

import org.junit.jupiter.api.Test;

import com.flexpoker.game.command.aggregate.TableBalancer;
import com.flexpoker.game.command.events.TableResumedAfterBalancingEvent;

public class TableBalancerResumedTableTest {

    @Test
    void testOneBalancedTableThatIsPaused() {
        var subjectTableId = UUID.randomUUID();
        var tableToPlayersMap = createTableToPlayersMap(subjectTableId, 2);

        var tableBalancer = new TableBalancer(UUID.randomUUID(), 2);
        var event = tableBalancer.createSingleBalancingEvent(subjectTableId, Collections.singleton(subjectTableId),
                tableToPlayersMap, createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap));
        assertEquals(TableResumedAfterBalancingEvent.class, event.get().getClass());
        assertEquals(subjectTableId, ((TableResumedAfterBalancingEvent) event.get()).getTableId());
    }

    @Test
    void testTwoBalancedTablesOnePaused() {
        var subjectTableId = UUID.randomUUID();
        var tableToPlayersMap = createTableToPlayersMap(subjectTableId, 8, 8);
        var otherTableId = tableToPlayersMap.keySet().stream()
                .filter(x -> !x.equals(subjectTableId)).findFirst().get();

        var tableBalancer = new TableBalancer(UUID.randomUUID(), 9);
        var event = tableBalancer.createSingleBalancingEvent(subjectTableId, Collections.singleton(otherTableId),
                tableToPlayersMap, createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap));
        assertEquals(TableResumedAfterBalancingEvent.class, event.get().getClass());
        assertEquals(otherTableId, ((TableResumedAfterBalancingEvent) event.get()).getTableId());
    }

}

File: src/test/kotlin/com/flexpoker/game/command/aggregate/tablebalancer/TableBalancerResumedTableTest.kt
package com.flexpoker.game.command.aggregate.tablebalancer

import com.flexpoker.game.command.aggregate.createSingleBalancingEvent
import com.flexpoker.game.command.events.TableResumedAfterBalancingEvent
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.UUID

class TableBalancerResumedTableTest {

    @Test
    fun testOneBalancedTableThatIsPaused() {
        val subjectTableId = UUID.randomUUID()
        val tableToPlayersMap = TableBalancerTestUtils.createTableToPlayersMap(subjectTableId, 2)
        val event = createSingleBalancingEvent(UUID.randomUUID(), 2, subjectTableId, setOf(subjectTableId),
            tableToPlayersMap, TableBalancerTestUtils.createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap))
        assertEquals(TableResumedAfterBalancingEvent::class.java, event.get().javaClass)
        assertEquals(subjectTableId, (event.get() as TableResumedAfterBalancingEvent).tableId)
    }

    @Test
    fun testTwoBalancedTablesOnePaused() {
        val subjectTableId = UUID.randomUUID()
        val tableToPlayersMap = TableBalancerTestUtils.createTableToPlayersMap(subjectTableId, 8, 8)
        val otherTableId = tableToPlayersMap.keys.first { it != subjectTableId }
        val event = createSingleBalancingEvent(UUID.randomUUID(), 9, subjectTableId, setOf(otherTableId),
            tableToPlayersMap, TableBalancerTestUtils.createDefaultChipMapForSubjectTable(subjectTableId, tableToPlayersMap))
        assertEquals(TableResumedAfterBalancingEvent::class.java, event.get().javaClass)
        assertEquals(otherTableId, (event.get() as TableResumedAfterBalancingEvent).tableId)
    }

}